R/as_curve_params.R

```diff
@@ -13,6 +13,20 @@
 #'
 #' print(curve_data)
 as_curve_params <- function(data, antigen_isos = NULL) {
+
+  if(!is.data.frame(data))
+  {
+    cli::cli_abort(
+      class = "not data.frame",
+      message = c(
+        "Can't convert {.arg data} to {.cls curve_params}.",
+        "x" = "{.arg data} must be a {.cls data.frame}
+        (or a subclass of {.cls data.frame}).",
+        "i" = "You have supplied a {.cls {class(data)}}."
+      )
+    )
+  }
```

Comment:
ideally we should use a more canned check like `assertthat::assert_that(is.data.frame(data))`, but that package seems abandoned since 2019 and doesn't use `cli` formatting, and I haven't found a more modern replacement yet.